pkg/search/postgres/common.go

```diff
@@ -589,6 +597,64 @@ func RunGetManyQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.
 	return results, nil
 }
 
+// RunCursorQueryForSchema creates a cursor against the database
+func RunCursorQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.Query, db *pgxpool.Pool) (fetcher func(n int) ([][]byte, error), closer func(), err error) {
+	query, err := standardizeQueryAndPopulatePath(q, schema, GET)
+	if err != nil {
+		return nil, nil, err
+	}
+	if query == nil {
+		return nil, nil, errox.InvalidArgs.New("empty query")
+	}
+
+	queryStr := query.AsSQL()
+
+	conn, err := db.Acquire(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	closer = func() {
+		conn.Release()
+	}
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		closer()
+		return nil, nil, err
+	}
+	closer = func() {
+		if err := tx.Rollback(ctx); err != nil {
```

Comment:
Why not commit? Why do we need rollback?

Reply:
yeah changed to commit

Reply:
Now the log statement is wrong